src/models/transfers.js

```diff
@@ -438,6 +438,7 @@ function * setTransfer (externalTransfer, requestingUser) {
         previousCredits, 'credit')
     }
 
+    yield db.upsertTransfer(transfer, {transaction})
```

Comment:
The transfer has to exist before holds can be created; I'll add a comment for this